controllers/tasksRequests.js

```diff
@@ -43,6 +46,73 @@ const fetchTaskRequestById = async (req, res) => {
   }
 };
 
+const addTaskRequests = async (req, res) => {
+  try {
+    const taskRequestData = req.body;
+    const usernamePromise = usersUtils.getUsername(taskRequestData.userId);
+    if (req.userData.id !== taskRequestData.userId && !req.userData.roles?.super_user) {
+      return res.boom.forbidden("Not authorized to create the request");
+    }
+    if (taskRequestData.proposedDeadline < taskRequestData.proposedStartDate) {
+      return res.boom.badRequest("Task deadline cannot be before the start date");
+    }
+    switch (req.body.requestType) {
+      case TASK_REQUEST_TYPE.ASSIGNMENT: {
```

Comment:
What ASSIGNMENT Type means??

Reply:
When task already exists and we request for it.